modules/instance_template/main.tf

```diff
@@ -146,9 +150,11 @@ resource "google_compute_instance_template" "tpl" {
   }
 
   scheduling {
-    preemptible         = var.preemptible
-    automatic_restart   = local.automatic_restart
-    on_host_maintenance = local.on_host_maintenance
+    preemptible                 = var.preemptible
+    automatic_restart           = local.automatic_restart
+    on_host_maintenance         = local.on_host_maintenance
+    provisioning_model          = local.provisioning_model
+    instance_termination_action = "STOP"
```

Comment:
One of the tests seem to be failing. It seems like we can only set instance_termination_action STOP if using spot. 
```
       
       Error: Error creating instance template: googleapi: Error 400: Invalid value for field 'resource.properties.scheduling.instanceTerminationAction': 'STOP'. You cannot specify a termination action for a VM instance that has the standard provisioning model (default). To use instance termination action, the VM instance must use the Spot provisioning model., invalid
       
         with module.instance_template_simple.module.instance_template.google_compute_instance_template.tpl,
         on ../../../../modules/instance_template/main.tf line 66, in resource "google_compute_instance_template" "tpl":
         66: resource "google_compute_instance_template" "tpl" {
         ```